File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        // as we need to find most frequent element we will keep this i pair 
        unordered_map<int,int> mp;
        for(int i=0;i<nums.size();i++){
            mp[nums[i]]++;
        }
        // Now elements wiht its frequency is stored.
        // top k frequent element
        priority_queue<pair<int,int>> pq;
        for(auto it: mp){
            pq.push({it.second,it.first});
        }
        vector<int> ans;
        while(k--){
            ans.push_back({pq.top().second});
            pq.pop();
        }

        return ans;
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        # first we will store everything in dictionary 
        mp= {}
        for i in nums:
            if i not in mp:
                mp[i]=1
            else:
                mp[i]+=1
        pq = []
        for i in mp:
            heapq.heappush(pq,(-mp[i],i))
        ans = []
        for i in range(0,k):
            ans.append(heapq.heappop(pq)[1])
        return ans